src/signaturev4.jl

```diff
@@ -0,0 +1,227 @@
+module SignatureV4
+
+using Base64
+using Dates
+using HTTP
+using HTTP.Pairs
+using HTTP.URIs
+using HTTP: Headers, Layer, Request
+using IniFile
+using MbedTLS
+
+export AWS4AuthLayer, sign!
+
+"""
+    AWS4AuthLayer{Next} <: HTTP.Layer
+
+Abstract type used by [`HTTP.request`](@ref) to add an
+[AWS Signature v4](http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html)
+authentication layer to the request.
+
+Historically this layer has been placed in-between the `MessageLayer` and `RetryLayer` in the HTTP
+stack. The request stack can be found
+[here](https://github.com/JuliaWeb/HTTP.jl/blob/v0.8.6/src/HTTP.jl#L534).
+
+An example of how to layer insert can be found below:
+```julia
+using AWSCore
+using HTTP
+
+stack = HTTP.stack()
+
+# This will insert the AWS4AuthLayer before the RetryLayer
+insert(stack, RetryLayer, AWS4AuthLayer)
+```
+"""
+abstract type AWS4AuthLayer{Next<:Layer} <: Layer{Next} end
+
+credentials = NamedTuple()
```

Comment:
Possibly remove? Globals are not very performant. If you really need to do this prefer:
```julia
const CREDENTIALS = Ref{NamedTuple}()

...
CREDENTIALS[] = NamedTuple()

```